neurodamus/node.py

```diff
@@ -1342,9 +1342,10 @@ def _coreneuron_ensure_all_ranks_have_gids(self, corenrn_data):
             return
 
         # create a fake node with a fake population "zzz" to get an unused gid.
-        # It only works with a single cycles without enough cells, but in
-        #  _instantiate_simulation we made sure we limit the number of cycles
+        # coreeuron fails if this edge case is reached multiple times as we
+        # try to add twice the same gid. pop "zzz" is forbidden
         log_verbose("Creating fake gid for CoreNeuron")
+        assert not PopulationNodes.get("zzz")
```

Comment:
With the fix of `max_cell_count`, I think we will add once the fake population `zzz`. Is this `assert` to protect in case a real population `zzz` exists ?

Reply:
yes. Do you think it is too much? The additional cost is negligible since we do this only at the beginning

Reply:
The assert is fine to be here. Maybe in the docstrings, `pop "zzz" is forbidden` => `pop "zzz" is forbidden if already existing`  ?

Reply:
Yes, it is ok to be brief. 
But what does the single sentence`pop "zzz" is forbidden` mean here?

Reply:
changed